File: Java/src/main/java/com/xurxodev/stringcalculator/NumberExtractorFromCommaSeparatedString.java
package com.xurxodev.stringcalculator;

import java.util.ArrayList;
import java.util.List;

public class NumberExtractorFromCommaSeparatedString implements StringCalculator.NumberExtractor {

    @Override
    public List<Integer> extract(String numbers) {
        List<Integer> numberList = new ArrayList<>();

        if (numbers != ""){
            for (String number:numbers.split(",")) {
                numberList.add(Integer.parseInt(number));
            }
        }

        return numberList;
    }
}

File: C#/StringCalculatorKata/NumberExtractorFromCommaSeparatedString.cs
using StringCalculatorKata.Boundaries;
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace StringCalculatorKata
{
    public class NumberExtractorFromCommaSeparatedString : INumberExtractor
    {
        public List<int> Extract(string numbers)
        {
            List<int> numberList = new List<int>();

            if (numbers != "")
            {
                foreach (string number in numbers.Split(','))
                {
                    numberList.Add(int.Parse(number));
                }
            }

            return numberList;
        }
    }
}
